syntax = "proto3";

package faker;

service Faker{
    rpc QuickFake(QuickFakeRequest) returns (QuickFakeResponse){}

    rpc BatchFake(BatchFakeRequest) returns (BatchFakeResponse){}
}

message QuickFakeRequest{
    string locale = 1;
    string provider = 2;
}

message QuickFakeResponse{
    string result = 1;
}

message ProviderMap{
    string key = 1;
    string value = 2;
}

message BatchFakeRequest{
    string locale = 1;
    int64 count = 2;
    repeated ProviderMap providerMaps = 3;
}

message BatchFakeResponse{
    string result = 1;
}